### ISSUE TITLE
onchange is not triggered when cancel a file input

### ISSUE THREAD

**Issue by wuv1982** [danialfarid/angular-file-upload#527]
first i selected a file, not use drag&drop, then i remove it by cancel a input .
 the `ng-model` value should be empty, but it didn't changed because `change` event is not triggered.

if i removed this part, the problem will be solved.
```javascript
if ($parse(attr.resetOnClick)(scope) != false) {
		if (navigator.appVersion.indexOf("MSIE 10") !== -1) {
			// fix for IE10 cannot set the value of the input to null programmatically by replacing input
			var replaceElem = function(evt) {
				var inputFile = elem.clone();
				inputFile.val('');
				elem.replaceWith(inputFile);
				$compile(inputFile)(scope);
				fileElem = inputFile;
				elem = inputFile;
				elem.bind('change', onChangeFn);
				elem.unbind('click');
				elem[0].click();
				elem.bind('click', replaceElem);
				evt.preventDefault();
				evt.stopPropagation();
			};
			elem.bind('click', replaceElem);
		} else {
			elem.bind('click', function(evt) {
				elem[0].value = null;
			});
		}
	}
```
i dont known whether there is anything wrong with here? 

your `Demo` has same problem, so you can reproduce it.

**Comment by danialfarid** [comment]
This part is to support cancel for browsers other than chrome. How removing it would make it work!

**Comment by wuv1982** [comment]
sorry for not give you more details. here is it.
### browser
Chrome Mac 40.0.2214.93 (64-bit)

### html
```html
<input type="file" ng-multiple="true" ng-file-select ng-model="upload.files" />
```

### js
```coffeescript
$scope.$watch 'upload.files', (n, o) ->
        $log.log "watch changed to: ", n
```

### screenshot
![2015-01-30 12 11 45](https://cloud.githubusercontent.com/assets/3183306/5970660/83437020-a879-11e4-9011-4bda59598449.png)

_canceled by Red Rectangle part in pic._
it gave me a log when i selected one file, but not gave anything when i canceled the `input`

**Comment by danialfarid** [comment]
What version of the plugin? Is it happening on the demo page too?

**Comment by wuv1982** [comment]
angular-file-upload : 2.2.2
angular : 1.3.11
---
In your demo, `$scope.generateThumb` is triggered when select ,but not triggered when cancel.

**Comment by danialfarid** [comment]
This should be fixed in version 3.0.0.
When you click on the file select the value is set to null so if the user cancels the popup and don't select another file your model is already set to null.

**Comment by wuv1982** [comment]
nice update. 
expect for the release.
